ebos/eclproblem.hh

```diff
@@ -2272,6 +2273,15 @@ private:
             const unsigned int elemIdx = elemMapper.index(element);
 
             elementCenterDepth_[elemIdx] = cellCenterDepth(element);
+            if (ecl_state.aquifer().hasNumericalAquifer()) {
+                const auto& num_aquifer = ecl_state.aquifer().numericalAquifers();
+                const unsigned int global_index = vanguard.cartesianIndex(elemIdx);
+                if (num_aquifer.hasCell(global_index)) {
+                    const auto& cell = num_aquifer.aquiferCells().at(global_index);
+                    elementCenterDepth_[elemIdx] = cell.depth;
+                }
+            }
+
         }
```

Comment:
I suggest we move the elementCenterDepth to the vanguard

See https://github.com/OPM/opm-simulators/pull/2975